test/hpke_test.c

```diff
@@ -0,0 +1,1432 @@
+    0x47, 0x92, 0xcd, 0x97, 0xba, 0x0e, 0xa1, 0xfa,
+    0xff, 0x09, 0xd9, 0x84, 0x35, 0x01, 0x23, 0x45,
+    0x76, 0x6e, 0xe3, 0x3a, 0xae, 0x2d, 0x76, 0x64
+};
+static const unsigned char fourth_ikmrpriv[] = {
+    0x33, 0xd1, 0x96, 0xc8, 0x30, 0xa1, 0x2f, 0x9a,
+    0xc6, 0x5d, 0x6e, 0x56, 0x5a, 0x59, 0x0d, 0x80,
+    0xf0, 0x4e, 0xe9, 0xb1, 0x9c, 0x83, 0xc8, 0x7f,
+    0x2c, 0x17, 0x0d, 0x97, 0x2a, 0x81, 0x28, 0x48
+};
+static const unsigned char fourth_expected_shared_secret[] = {
+    0xe8, 0x17, 0x16, 0xce, 0x8f, 0x73, 0x14, 0x1d,
+    0x4f, 0x25, 0xee, 0x90, 0x98, 0xef, 0xc9, 0x68,
+    0xc9, 0x1e, 0x5b, 0x8c, 0xe5, 0x2f, 0xff, 0xf5,
+    0x9d, 0x64, 0x03, 0x9e, 0x82, 0x91, 0x8b, 0x66
+};
+static const unsigned char fourth_export1[] = {
+    0x7a, 0x36, 0x22, 0x1b, 0xd5, 0x6d, 0x50, 0xfb,
+    0x51, 0xee, 0x65, 0xed, 0xfd, 0x98, 0xd0, 0x6a,
+    0x23, 0xc4, 0xdc, 0x87, 0x08, 0x5a, 0xa5, 0x86,
+    0x6c, 0xb7, 0x08, 0x72, 0x44, 0xbd, 0x2a, 0x36
+};
+static const unsigned char fourth_context2[] = { 0x00 };
+static const unsigned char fourth_export2[] = {
+    0xd5, 0x53, 0x5b, 0x87, 0x09, 0x9c, 0x6c, 0x3c,
+    0xe8, 0x0d, 0xc1, 0x12, 0xa2, 0x67, 0x1c, 0x6e,
+    0xc8, 0xe8, 0x11, 0xa2, 0xf2, 0x84, 0xf9, 0x48,
+    0xce, 0xc6, 0xdd, 0x17, 0x08, 0xee, 0x33, 0xf0
+};
+static const unsigned char fourth_context3[] = {
+    0x54, 0x65, 0x73, 0x74, 0x43, 0x6f, 0x6e, 0x74,
+    0x65, 0x78, 0x74
+};
+static const unsigned char fourth_export3[] = {
+    0xff, 0xaa, 0xbc, 0x85, 0xa7, 0x76, 0x13, 0x6c,
+    0xa0, 0xc3, 0x78, 0xe5, 0xd0, 0x84, 0xc9, 0x14,
+    0x0a, 0xb5, 0x52, 0xb7, 0x8f, 0x03, 0x9d, 0x2e,
+    0x87, 0x75, 0xf2, 0x6e, 0xff, 0xf4, 0xc7, 0x0e
+};
+
+static int export_only_test(void)
+{
+    /* based on RFC9180 A.7 */
+    const TEST_BASEDATA basedata = {
+        OSSL_HPKE_MODE_BASE,
+        {
+            OSSL_HPKE_KEM_ID_X25519,
+            OSSL_HPKE_KDF_ID_HKDF_SHA256,
+            OSSL_HPKE_AEAD_ID_EXPORTONLY
+        },
+        fourth_ikme, sizeof(fourth_ikme),
+        fourth_ikmepub, sizeof(fourth_ikmepub),
+        fourth_ikmr, sizeof(fourth_ikmr),
+        fourth_ikmrpub, sizeof(fourth_ikmrpub),
+        fourth_ikmrpriv, sizeof(fourth_ikmrpriv),
+        fourth_expected_shared_secret, sizeof(fourth_expected_shared_secret),
+        ksinfo, sizeof(ksinfo),
+        NULL, 0, /* no auth */
+        NULL, 0, NULL /* PSK stuff */
+    };
+    const TEST_EXPORTDATA exportdata[] = {
+        { NULL, 0, fourth_export1, sizeof(fourth_export1) },
+        { fourth_context2, sizeof(fourth_context2),
+          fourth_export2, sizeof(fourth_export2) },
+        { fourth_context3, sizeof(fourth_context3),
+          fourth_export3, sizeof(fourth_export3) },
+    };
+    return do_testhpke(&basedata, NULL, 0,
+                       exportdata, OSSL_NELEM(exportdata));
+}
+
+/*
+ * Randomly toss a coin
+ */
+static unsigned char rb = 0;
+#define COIN_IS_HEADS (RAND_bytes_ex(testctx, &rb, 1, 10) && rb % 2)
```

Comment:
Use `test_random()` instead. The test harness will display the seed used during the test, and it is possible to re-run the test with that SEED giving repeatable results.

Reply:
nice! hadn't seen that one before